File: frontend/src/services/organizationService.ts
import api from './api';

export interface Organization {
  id: string;
  name: string;
  parentId?: string;
  type: string;
}

export const organizationService = {
  getAll: async (): Promise<Organization[]> => {
    const response = await api.get('/organizations');
    return response.data;
  },

  getTree: async (): Promise<Organization[]> => {
    const response = await api.get('/organizations/tree');
    return response.data;
  },

  getById: async (id: string): Promise<Organization> => {
    const response = await api.get(`/organizations/${id}`);
    return response.data;
  },

  create: async (data: Partial<Organization>): Promise<Organization> => {
    const response = await api.post('/organizations', data);
    return response.data;
  },

  update: async (id: string, data: Partial<Organization>): Promise<Organization> => {
    const response = await api.put(`/organizations/${id}`, data);
    return response.data;
  },

  delete: async (id: string): Promise<void> => {
    await api.delete(`/organizations/${id}`);
  },
};

File: backend/app/services/organizationService.py
from typing import List, Optional
from sqlalchemy.orm import Session

from ..models.organization import Organization
from ..repositories.organizationRepository import OrganizationRepository


class OrganizationService:
    def __init__(self, db: Session):
        self.db = db
        self.repository = OrganizationRepository(db)

    def get_all(self) -> List[Organization]:
        pass

    def get_tree(self) -> List[Organization]:
        pass

    def get_by_id(self, org_id: str) -> Optional[Organization]:
        pass

    def create(self, org_data: dict) -> Organization:
        pass

    def update(self, org_id: str, org_data: dict) -> Optional[Organization]:
        pass

    def delete(self, org_id: str) -> bool:
        pass

    def get_children(self, parent_id: str) -> List[Organization]:
        pass
